src/server.cpp

```diff
@@ -2257,6 +2304,17 @@ void Server::fadeSound(s32 handle, float step, float gain)
 		m_playing_sounds.erase(it);
 }
 
+void Server::createSoundPacket(NetworkPacket &pkt, s32 sound_id, const ServerPlayingSound &params, const v3f &pos, bool ephemeral)
+{
+	float gain = params.gain * params.spec.gain;
+	pkt << sound_id << params.spec.name << gain
+			<< (u8) params.type << pos << params.object
```

Comment:
`ServerPlayingSound` already contains a `pos` member which could be used instead of carrying around the `pos` function parameter.

Reply:
It means multiple calling of `ServerPlayingSound::getPos` because pos is also needed for different purposes in both places where `createSoundPacket` is called.
Is it better to add an extra call instead of using reference in method parameters?

Reply:
There are the `pos` and `object` fields. It depends on position type if `pos` is read from `pos` or `object`.

So it looks better to carry it by parameter and reuse the already taken position returned by the `getPos` method.